### PR TITLE
RelatedPerson Patch Family Name

### PR DESCRIPTION
**Pull request by SantanuBanerjee**
Description
----

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

Screenshot 2021-01-05 at 4 52 18 PM

Screenshot 2021-01-05 at 4 52 03 PM

### CODE REVIEW

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 1]
```diff
@@ -777,7 +777,17 @@ module Resources
             'start': '2012-05-17T15:33:18.000Z'
           }
         }
-      }
+      },
+      {
+        'path': '/name/0/id',
```
Comment (sydneanschutz) [inline lib/resources/example_json/r4_examples_relatedperson.rb:782]: just to be consistent with the rest of the examples on this resource, can we put the op first and then the path. on all your examples

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 2]
```diff
@@ -777,7 +777,17 @@ module Resources
...
+        'op': 'test',
+        'value': 'CI-12467982-0'
```
Comment (sydneanschutz) [inline lib/resources/example_json/r4_examples_relatedperson.rb:784]: I dont think this is real id for name? it should look like 'CI-12724068-12724066-0' Thats the example in laurens read example, you should be able to use it

File: lib/resources/r4/related_person_patch.yaml [hunk 3]
```diff
@@ -64,3 +64,37 @@ operations:
         <li>Must include a time component with a timezone when specifying a `period`.</li>
         <li>If `line` is provided, only the first four values are used. Additional values will be ignored.</li>
       </ul>
+
+  - name: test-name-id
+    path: /name/{index}/id
+    operation: test
+    type: string
+    url: https://hl7.org/fhir/R4/element-definitions.html#Element.id
```
Comment (sydneanschutz) [inline lib/resources/r4/related_person_patch.yaml:72]: I think patient test operations are linking to a fhir string link... https://hl7.org/fhir/R4/datatypes.html#string
do we want element.id? im not sure

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 1 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb]: Changed in all.
https://github.com/cerner/fhir.cerner.com/commit/d6ef4e31556be3decb0db09531e1e82211b51d12#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eR783

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 2 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb:784]: https://github.com/cerner/fhir.cerner.com/commit/d6ef4e31556be3decb0db09531e1e82211b51d12#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR78

File: lib/resources/r4/related_person_patch.yaml [hunk 4]
```diff
@@ -64,3 +64,37 @@ operations:
...
+    description: Test the value of the `HumanName.id` of the name at the given <code>{index}</code> in the list of related person names.
+    example: |
+      {
+        "path": "/name/0/id",
```
Comment (sydneanschutz) [inline lib/resources/r4/related_person_patch.yaml:76]: op then path here too please

File: lib/resources/r4/related_person_patch.yaml [hunk 5]
```diff
@@ -64,3 +64,37 @@ operations:
...
+        "value": "CI-12724068-12724066-0"
+      }
+    note: >
+      <ul>
+        <li>This operation is only supported for the first name in the list of related person names (<code>{index}</code> == 0).</li>
+      </ul>
+  - name: replace-name-family
+    path: /name/{index}/family
+    operation: replace
+    type: String
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:87]: I believe this needs to be lowercase 'string' to complete the hyperlink.

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 6]
```diff
@@ -777,7 +777,17 @@ module Resources
             'start': '2012-05-17T15:33:18.000Z'
           }
         }
-      }
+      },
+      {
+        'op': 'test',
+        'path': '/name/0/id',
+        'value': 'CI-12724068-12724066-0'
```
Comment (sydneanschutz) [inline lib/resources/example_json/r4_examples_relatedperson.rb:784]: Gonna have o put these back, but make sure it's a valid name id in C1941. Can you also fix the name id examples in else where in this file, for example: https://github.com/cerner/fhir.cerner.com/blob/main/lib/resources/example_json/r4_examples_relatedperson.rb#L91

File: lib/resources/r4/related_person_patch.yaml (hunk 4 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/related_person_patch.yaml]: https://github.com/cerner/fhir.cerner.com/commit/c9bce63c67bb7f03d1c33b644945dfaa15d064db#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR77

File: lib/resources/r4/related_person_patch.yaml (hunk 5 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/related_person_patch.yaml]: https://github.com/cerner/fhir.cerner.com/commit/c9bce63c67bb7f03d1c33b644945dfaa15d064db#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR87

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 6 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb]: Changed the value here and other places from your prod validation jira, OPENSVC-15683. Is this fine?
Is there any other place I am missing, where the id needs to be replaced?

https://github.com/cerner/fhir.cerner.com/commit/c9bce63c67bb7f03d1c33b644945dfaa15d064db#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eR784

**Comment by SantanuBanerjee** [comment]
closing it for now, as it is not ready to go now for a dependency.

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 7]
```diff
@@ -88,7 +88,7 @@ module Resources
       ],
       'name': [
         {
-          'id': 'CI-12724068-12724066-0',
+          'id': 'CI-12782404-0',
```
Comment (sydneanschutz) [inline lib/resources/example_json/r4_examples_relatedperson.rb:91]: lauren will be changing this, so maybe rebase off of her after she merges https://github.com/cerner/fhir.cerner.com/pull/540/files#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eL91

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 8]
```diff
@@ -777,7 +777,17 @@ module Resources
             'start': '2012-05-17T15:33:18.000Z'
           }
         }
-      }
+      },
+      {
+        'op': 'test',
+        'path': '/name/0/id',
+        'value': 'CI-12782404-0'
```
Comment (sydneanschutz) [inline lib/resources/example_json/r4_examples_relatedperson.rb:784]: the example patch is calling for RP E-12467982-97731398
so i think the id should be CI-12467982-0

File: lib/resources/r4/related_person_patch.yaml (hunk 5 above)
Comment (exopsRamba) [reply lib/resources/r4/related_person_patch.yaml]: Do you mind posting a screenshot showing the hyperlink works now?

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 8 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb]: https://github.com/cerner/fhir.cerner.com/commit/8bbec6c06cc084c792e14d3c466bf4f299336cfa#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eR786

File: lib/resources/r4/related_person_patch.yaml (hunk 5 above)
Comment (SantanuBanerjee) [reply lib/resources/r4/related_person_patch.yaml]: Screenshot 2021-01-21 at 5 51 27 PM
Screenshot 2021-01-21 at 5 51 41 PM

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 7 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb]: rebased.

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 2 above)
Comment (SantanuBanerjee) [reply lib/resources/example_json/r4_examples_relatedperson.rb:786]: Responded in the other comment.

**Review by sydneanschutz — APPROVED** [review]

**Review by exopsRamba — APPROVED** [review]

**Review by alexbeckner1 — APPROVED** [review]

**Comment by sydneanschutz** [comment]
Verified on live: 
Screen Shot 2021-01-21 at 2 04 10 PM

**Pull request merged**